plugins/e57/io/Utils.cpp

```diff
@@ -211,27 +217,15 @@ bool getLimits(const e57::StructureNode& prototype,
     return true;
 }
 
+// This will give positive bounds to default data type for id.
 std::pair<uint64_t, uint64_t> getPdalBounds(pdal::Dimension::Id id)
 {
-    //ABELL - This is strange.  PDAL doesn't specify limits for these.
-    //  Perhaps argument registration needs to be changed.
-    using Dim = pdal::Dimension::Id;
-    switch (id)
-    {
-        case Dim::Red:
-        case Dim::Blue:
-        case Dim::Green:
-        case Dim::Intensity:
-            return {std::numeric_limits<uint16_t>::min(),
-                    std::numeric_limits<uint16_t>::max()};
-        case Dim::Classification:
-            return {std::numeric_limits<uint8_t>::min(),
-                    std::numeric_limits<uint8_t>::max()};
-        default:
-            std::string msg ="Dimension " + Dimension::name(id) +
-                             " is not currently supported.";
-            throw pdal_error(msg);
-    }
+    // pdal::Dimension::size() returns number of bytes for the pdal::Dimesion::Type.
+    // eg: 1 for uint8, 2 for uint16, 4 for uint32, 8 for double, etc.
+    // Max range for data type = (2 ^ (8 * no. of bytes)) - 1
+    auto type = pdal::Dimension::defaultType(id);
+    auto maxVal = std::pow(2, 8 * pdal::Dimension::size(type)) - 1;
+    return {0, maxVal};
```

Comment:
Nice one! Could you throw an exception if the type is signed or not integer?

Reply:
Done